src/galileo/prompts.py

```diff
@@ -127,4 +264,159 @@ def get_prompt_template(project: str, name: str) -> Optional[PromptTemplate]:
 
 
 def list_prompt_templates(project: str) -> list[PromptTemplate]:
+    warnings.warn("list_prompt_templates is deprecated, use get_prompts instead.", DeprecationWarning, stacklevel=2)
     return PromptTemplates().list(project_name=project)
+
+
+@overload
+def get_prompt(*, name: str, version: int) -> Optional[PromptTemplateVersion]: ...
+
+
+@overload
+def get_prompt(*, template_id: str) -> Optional[PromptTemplateVersion]: ...
+
+
+@overload
+def get_prompt(*, name: str) -> Optional[PromptTemplateVersion]: ...
+
+
+def get_prompt(
+    *, template_id: Optional[str] = None, name: Optional[str] = None, version: Optional[int] = None
+) -> Optional[PromptTemplateVersion]:
+    """
+    Retrieves a specific global prompt template version.
+
+    You must provide either 'template_id' or 'name', but not both.
+    If 'version' is not provided, the currently selected version is returned.
+    Parameters
+    ----------
+    template_id : str, optional
+        The unique identifier of the template to retrieve. Defaults to None.
+    name : str, optional
+        The name of the template to retrieve. Defaults to None.
+    version : int, optional
+        The version number to retrieve. If not provided, the currently selected version is returned. Defaults to None.
+    Returns
+    -------
+    Optional[PromptTemplateVersion]
+        The template version if found, None otherwise.
+    Raises
+    ------
+    ValueError
+        If neither or both 'template_id' and 'name' are provided.
+    """
+    if (template_id is None) and (name is None):
+        raise ValueError("Exactly one of 'template_id' or 'name' must be provided")
+
+    if (template_id is not None) and (name is not None):
+        raise ValueError("Exactly one of 'template_id' or 'name' must be provided")
+
+    if version is not None:
+        target_id = template_id
+        if name:
+            parent_template = GlobalPromptTemplates().get(name=name)
+            if not parent_template:
+                return None
+            target_id = parent_template.id
+
+        if not target_id:
+            raise ValueError("A template id is required to fetch a specific version.")
+
+        return GlobalPromptTemplates().get_version(template_id=target_id, version=version)
+
+    prompt_template: Optional[PromptTemplate] = None
```

Comment:
Considering that you are already validating that either the id or the name should be present you can simplify this code to just:

```suggestion
    if template_id:
        prompt_template = GlobalPromptTemplates().get(template_id=template_id)
    else:
        prompt_template = GlobalPromptTemplates().get(name=name)
```